src/main/java/org/dcsa/ebl/service/impl/ShippingInstructionTOServiceImpl.java

```diff
@@ -192,7 +192,10 @@
                     return extractShipmentRelatedFields(shippingInstructionTO, shipmentIds, tuples)
                             .then(Mono.just(cargoItemTOs));
                 })
-                .doOnNext(shippingInstructionTO::setCargoItems)
+                .doOnNext(cargoItemTOs -> {
+                    shippingInstructionTO.setCargoItems(cargoItemTOs);
+                    shippingInstructionTO.hoistCarrierBookingReferenceIfPossible();
```

Comment:
Don't you need to catch the IllegalStateException here as well as when you do the pushCarrierBookingRe...

Reply:
I did not bother because it is an internal error situation. :)

Reply:
(the other case is processing the input, so it is about giving the caller a good error message where this is a "Should not happen")